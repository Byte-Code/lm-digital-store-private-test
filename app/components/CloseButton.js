import React, { PropTypes } from 'react';
import BlockIcon from 'material-ui/svg-icons/navigation/close';

const CloseButton = ({ handleClick, top, right, backgroundColor, fill }) => (
  <BlockIcon
    style={{
      fill,
      backgroundColor,
      top,
      right,
      width: 80,
      height: 80,
      position: 'absolute',
      cursor: 'pointer'
    }}
    color="#fff"
    onTouchTap={
      (e) => {
        if (e) {
          e.preventDefault();
        }
        handleClick();
      }
    }
  />
);

CloseButton.propTypes = {
  handleClick: PropTypes.func.isRequired,
  top: PropTypes.number,
  right: PropTypes.number,
  fill: PropTypes.string,
  backgroundColor: PropTypes.string
};

CloseButton.defaultProps = {
  top: 0,
  right: 0,
  fill: '#858585',
  backgroundColor: '#333333'
};

export default CloseButton;
