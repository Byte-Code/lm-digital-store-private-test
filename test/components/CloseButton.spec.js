import React from 'react';
import { shallow } from 'enzyme';
import BlockIcon from 'material-ui/svg-icons/navigation/close';

import CloseButton from '../../app/components/CloseButton';

describe('CloseButton', () => {
  const handleClick = jest.fn();

  it('should render properly when no styles are provided', () => {
    const result = shallow(
      <CloseButton handleClick={handleClick} />
    );
    expect(result).toMatchSnapshot();
  });

  it('should render properly when styles are provided', () => {
    const result = shallow(
      <CloseButton
        handleClick={handleClick}
        top={10}
        right={10}
        fill="#fff"
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('should call handleClick when tbe Icon is clicked', () => {
    const result = shallow(
      <CloseButton handleClick={handleClick} />
    );
    result.find(BlockIcon).simulate('touchTap');
    expect(handleClick).toHaveBeenCalled();
  });
});
