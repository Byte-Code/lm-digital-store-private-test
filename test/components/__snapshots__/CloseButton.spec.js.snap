// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton should render properly when no styles are provided 1`] = `
<NavigationClose
  color="#fff"
  onTouchTap={[Function]}
  style={
    Object {
      "backgroundColor": "#333333",
      "fill": "#858585",
      "height": 80,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "width": 80,
    }
  }
/>
`;

exports[`CloseButton should render properly when styles are provided 1`] = `
<NavigationClose
  color="#fff"
  onTouchTap={[Function]}
  style={
    Object {
      "backgroundColor": "#333333",
      "fill": "#fff",
      "height": 80,
      "position": "absolute",
      "right": 10,
      "top": 10,
      "width": 80,
    }
  }
/>
`;
